controllers/webhook/storageclass.go

```diff
@@ -38,7 +43,86 @@ func (s *StorageClassAdmission) Handle(ctx context.Context, req admission.Reques
 		return admission.Errored(http.StatusBadRequest, fmt.Errorf("supported provisioners are %s", supportedProvisioners))
 	}
 
-	// TODO: mutate the storageclass with default parameters based on info from storagecluster
+	storageConsumerList := &v1alpha1.StorageConsumerList{}
+	if err := s.List(ctx, storageConsumerList, client.InNamespace(s.Namespace)); err != nil {
+		s.Log.Error(err, "failed to list storageconsumers", "namespace", s.Namespace)
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed to list storageconsumers in %s namespace", s.Namespace))
+	}
+
+	if len(storageConsumerList.Items) == 0 {
+		s.Log.Error(fmt.Errorf("no storageconsumers found in %s namespace", s.Namespace), "failed validation")
+		return admission.Denied("creation of storageclass should happen after storageconsumer exists")
+	}
+
+	clusterID := ""
+	for idx := range storageConsumerList.Items {
+		consumer := &storageConsumerList.Items[idx]
+		if consumer.Annotations[defaults.StorageConsumerTypeAnnotation] == defaults.StorageConsumerTypeLocal {
```

Comment:
what happens if i create a storageconsumer and add local annotation. no one will block me from doing that isnt it?

Reply:
yes, either this or an immutable field in the spec, we went w/ the annotation as not to backport spec changes.

let's discuss in upcoming meetings if any change is required.